haystack/schema.py

```diff
@@ -503,6 +508,7 @@ def __init__(self,
         self.labels = labels
 
         self.query = self._aggregate_labels(key="query", must_be_single_value=True)[0]
+        self.filters = self._aggregate_labels(key="filters", must_be_single_value=True)[0]
```

Comment:
This assumes we aggregate the labels always by filters like setting `aggregate_by_meta` in `BaseDocumentStore.get_all_labels_aggregated()` but which works on label's meta actually, however. Looks like the meta field's intention overlaps with the new filters field. And I can see cases where we do want to simply ignore the filters in the labels: If we have user feedback with filters, that doesn't actually mean that the label is only correct within the filter context but often might also be correct without the filter.

Reply:
Whenever we create a `MultiLabel` out of multiple `Label`s, we will aggregate their `filters` and in case not all `Label`s have the same `filters` there will be an error message. That way, filters are handled in the same way as queries during the aggregation. `get_all_labels_aggregated` will trigger that aggregation and create a `MultiLabel` form `Label`s, yes.

In case we have user feedback with filters, it could be that the label is also correct without the filters. However, we don't know that so I'd say we can't ignore the filters.

Reply:
Regarding the two fields filters and meta, they could both be joined into meta yes. I'd say the semantics are different. For example, meta can hold the annotator's name while filters is strongly connected to the query. While filters can be aggregated over multiple Labels on the same question, the annotator's name or any other info within meta cannot be aggregated.

Reply:
Ok.